rosidl_typesupport_opensplice_cpp/include/rosidl_typesupport_opensplice_cpp/Requester.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef __rosidl_typesupport_opensplice_cpp__Requester__h__
+#define __rosidl_typesupport_opensplice_cpp__Requester__h__
+
+#include <atomic>
+#include <random>
+#include <limits>
+#include <ccpp_dds_dcps.h>
+#include <u_entity.h>
+
+#include <rosidl_typesupport_opensplice_cpp/message_type_support.h>
+#include <rosidl_typesupport_opensplice_cpp/service_type_support.h>
+
+namespace rosidl_typesupport_opensplice_cpp
+{
+
+template<typename RequestT, typename ResponseT>
+class Requester
+{
+  public:
+    Requester(DDS::DomainParticipant_ptr participant, const char * service_name,
+              const char * service_type_name) :
```

Comment:
The class is immediately storing the incoming `const char *` in a `std::string` class member variable. So in this case (AFAIU) changing the parameter will not have any effect on performance.

But in general I would prefer the use of `std::string` simply to be idiomatic to C++ and only in exceptional cases where a performance gain can be made (and needs to be made) use the `const char *`. Other reasons to use `const char *` are if you're going to be coming or going through a C API. I guess what I'm getting at is that I would prefer to default to `std::string` unless there is a good reason to not use it, rather than the other way around. Also, a lot of libraries made the decision to avoid the overhead of STL containers before the move semantics and rvalue's were introduced, so it might not be the best reason to adopt the same policy.